lib/AbiBinProvider.js

```diff
@@ -1,111 +1,112 @@
+      return this.custom[contractName].abi;
     }
 
-    if (!abi) {
-      // We did not find abi in our location.
-      // Lets get it from mosaicAbiBinProvider.
-      return oThis.mosaicAbiBinProvider.getABI(contractName);
+    if (btContracts[contractName] && btContracts[contractName].abi) {
+      return btContracts[contractName].abi;
     }
-
-    return abi;
+    throw new Error(
+      `Could not retrieve ABI for ${contractName}.`,
+    );
   }
 
   /**
    * Getter to get BIN for a contract.
-   * @param contractName Name of the contract.
-   * @returns {String} Binary string.
+   *
+   * @param {string} contractName Name of the contract.
+   *
+   * @returns {string} Binary string.
    */
   getBIN(contractName) {
-    const oThis = this;
-    let bin = null;
-    try {
-      bin = super.getBIN(contractName);
-    } catch (e) {
-      // Just catch the exception. Do nothing.
+    if (this.custom[contractName] && this.custom[contractName].bin) {
+      return this.custom[contractName].bin;
     }
 
-    if (!bin) {
-      // We did not find abi in our location.
-      // Lets get it from mosaicAbiBinProvider.
-      return oThis.mosaicAbiBinProvider.getBIN(contractName);
+    if (btContracts[contractName] && btContracts[contractName].bin) {
+      return btContracts[contractName].bin;
     }
-    return bin;
+    throw new Error(
+      `Could not retrieve BIN for ${contractName}.`,
+    );
+  }
+
+  /**
+   * Method to add a custom abi for a contract.
+   *
+   * @param {string} contractName Name of the contract.
+   * @param {string|Object} contractAbi ABI of the contract.
+   */
+  addABI(contractName, contractAbi) {
+    let abi;
+
+    if (typeof contractAbi === 'string') {
+      abi = JSON.parse(contractAbi);
+    } else if (typeof contractAbi === 'object') {
+      abi = contractAbi;
+    } else {
+      throw new Error(
+        `Couldn't add ABI for contract ${contractName}, content should be string or object`,
+      );
+    }
+
+    if (!this.custom[contractName]) {
+      this.custom[contractName] = {};
+    }
+    const contract = this.custom[contractName];
+
+    if (!contract.abi) {
+      contract.abi = abi;
+    } else {
+      throw new Error(
+        `Abi for Contract Name ${contractName} already exists.`,
+      );
+    }
+  }
+
+  /**
+   * Method to add a custom bin for a contract.
+   *
+   * @param {string} contractName Name of the contract.
+   * @param {string} contractBin Bin of the the contract.
+   */
+  addBIN(contractName, contractBin) {
+    if (typeof contractBin !== 'string') {
+      throw new Error('Bin should be a string');
+    }
+
+    if (!this.custom[contractName]) {
+      this.custom[contractName] = {};
+    }
+    const contract = this.custom[contractName];
+
+    if (contract.bin) {
+      throw new Error(
+        `Bin for Contract Name ${contractName} already exists.`,
+      );
+    }
+
+    contract.bin = contractBin;
```

Comment:
You could probably refactor the `addABI` and `addBIN` methods to call on something like `addCustom('abi', contractName, contractAbi)` (similar for bin).